tests/Cache/FileCacheTest.php

```diff
@@ -76,11 +76,20 @@ public function testBrokenCacheRead(): void
         // Now simulate the file being corrupted and attempt to read it
         file_put_contents($files[0], '');
 
-        $this->expectException(CacheException::class);
+        $this->expectExceptionObject(new CacheException('Can not load cache for a feature "broken_feature" from '));
```

Comment:
I find this use of `expectExceptionObject` (and the one below) a little confusing : on first read it suggests that the caught exception should be equal to the passed one, but in fact it is only looking for a partial match on the exception message.

For cases like this I think it's clearer to expect the class and partial message separately?

Reply:
I did actually get confused when I changed the message globally by mistake, but tests still passed fine.

Then I found that `expectExceptionObject()` simply calls `expectExceptionMessage()` - and in fact it has always been a substring match!! Very confusing, but the code doesn't lie. :)

It worries me more that the existing tests are not checking the exact message string, which is mainly where we want this functionality. I can use `expectExceptionMessageRegExp()` for this case, but it feels like it just spreading a lie (or at least a misconception). 🙈